### PR TITLE
Header component

### PR DESCRIPTION
**Pull request by auqibyousuf**

### CODE REVIEW

File: app/components/NavBar/NavBar.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import { NavBarType } from "./NavBarTypes";
+import LinkComponent from "../Link/Link";
+import classNames from "classnames";
+
+const NavBar = ({ menuLinks, extraClasses }: NavBarType) => {
+  const listContainerClasses = classNames("flex relative");
+  const listItemClasses = classNames("group p-2 pr-5 relative");
```
Comment (UmmarHamid) [inline app/components/NavBar/NavBar.tsx]: instead of padding , can we use gap-2 on the wrapper element?

File: app/components/NavBar/NavBar.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  return (
+    <ul className={listContainerClasses}>
+      {menuLinks.map((linkItem) => {
+        const { text, url } = linkItem;
+        return (
+          <li key={text} className={listItemClasses}>
```
Comment (UmmarHamid) [inline app/components/NavBar/NavBar.tsx]: lets use the index here as some links may have same text

File: app/components/NavBar/NavBarTypes.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { LinkType } from "../Link/LinkTypes";
+
+export type NavBarType = {
+  menuLinks: LinkType[];
```
Comment (UmmarHamid) [inline app/components/NavBar/NavBarTypes.ts]:
```suggestion
  menuLinks: Array<LinkType>;
```

File: app/components/Header/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    "bg-darkBlue h-28 py-3 w-full flex justify-center items-center"
+  );
+  const headerChildClasses = classNames("");
+  const iconClasses = classNames(
+    "text-white w-5 h-5 pointer fill-white ml-auto"
+  );
+
+  return (
+    <header className={`${headerClasses} ${extraClasses ?? ""}`}>
+      <div className={headerChildClasses}>
```
Comment (UmmarHamid) [inline app/components/Header/Header.tsx]: can we remove this element and combine the classes on the nav element?

File: app/components/Header/Header.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+const Header = ({ logo, menu, ActionIcon, extraClasses }: HeaderType) => {
+  const navBarClasses = classNames(
+    "flex gap-4 items-center w-[1440px] px-10 h-full flex mx-auto"
+  );
+  const headerClasses = classNames(
+    "bg-darkBlue h-28 py-3 w-full flex justify-center items-center"
+  );
+  const headerChildClasses = classNames("");
+  const iconClasses = classNames(
+    "text-white w-5 h-5 pointer fill-white ml-auto"
```
Comment (UmmarHamid) [inline app/components/Header/Header.tsx]: should it use the blueish color instead?

File: app/components/Header/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import { HeaderType } from "./HeaderTypes";
+import classNames from "classnames";
+import NavBar from "../NavBar/NavBar";
+import { Logo } from "../Logo/Logo";
+
+const Header = ({ logo, menu, ActionIcon, extraClasses }: HeaderType) => {
+  const navBarClasses = classNames(
+    "flex gap-4 items-center w-[1440px] px-10 h-full flex mx-auto"
```
Comment (UmmarHamid) [inline app/components/Header/Header.tsx]: we can use max-width instead

**Review by UmmarHamid — CHANGES_REQUESTED** [review]
Looks great!.

Just small nit picks

File: app/components/NavBar/NavBar.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        return (
+          <li
+            key={text}
+            className={`${listItemClasses} ${listItemExtraClasses ?? ""}`}
+          >
+            <LinkComponent url={url} text={text} />
+          </li>
+        );
+      })}
+    </ul>
```
Comment (UmmarHamid) [inline app/components/NavBar/NavBar.tsx]:
```suggestion
      {menuLinks.map((linkItem, index) => {
        const { text, url } = linkItem;
        return (
          <li
            key={index}
            className={`${listItemClasses} ${listItemExtraClasses ?? ""}`}
          >
            <LinkComponent url={url} text={text} />
          </li>
        );
      })}
    </ul>
```

**Review by UmmarHamid — CHANGES_REQUESTED** [review]

File: app/components/NavBar/NavBar.tsx (hunk 2 above)
Comment (UmmarHamid) [reply app/components/NavBar/NavBar.tsx]: @auqibyousuf you missed this one?

Comment (auqibyousuf) [reply app/components/NavBar/NavBar.tsx]: done, was showing error

**Comment by auqibyousuf** [comment]
Done everything

**Review by UmmarHamid — APPROVED** [review]

**Pull request merged**